tests/proofs/functions-spec.k

```diff
@@ -0,0 +1,17 @@
+requires "kwasm-lemmas.k"
+
+module FUNCTIONS-LEMMAS
+    imports KWASM-LEMMAS
+endmodule
+
+module FUNCTIONS-SPEC
+    imports FUNCTIONS-LEMMAS
+
+    rule <k> #wrap(2, #getRange(ByteMap <| BM0 |>, ADDR, 4)) => #getRange(ByteMap <| BM0 |>, ADDR, 2) ... </k>
+      requires #isByteMap(ByteMap <| BM0 |>)
+       andBool 0 <=Int ADDR
+
+    rule <k> #setRange ( ByteMap <| BM0 |> , ADDR , #wrap ( 2 , #getRange ( ByteMap <| BM0 |> , ADDR , 4 ) ) , 2 ) => ByteMap <| BM0 |> ... </k>
+      requires #isByteMap(ByteMap <| BM0 |>)
+       andBool 0 <=Int ADDR
```

Comment:
You don't need to expose the constructor here, just let the map be symbolic. Then you also don't need to change this when we switch to `Bytes`.

Also, one proof used constructor-like formatting (`#getRange ( A , B , C )`) and another function-like formatting (`#getRange(A, B, C)`). I switched to the latter here, but I'm okay with either.

```suggestion
    rule <k> #wrap(2, #getRange(BM, ADDR, 4)) => #getRange(BM, ADDR, 2) ... </k>
      requires #isByteMap(BM)
       andBool 0 <=Int ADDR

    rule <k> #setRange(BM, ADDR, #wrap(2, #getRange(BM, ADDR, 4)), 2) => BM ... </k>
      requires #isByteMap(BM)
       andBool 0 <=Int ADDR
```

Reply:
Fixed.